src/MatrixFields/single_field_solver.jl

```diff
@@ -80,6 +80,57 @@ single_field_solve_kernel!(
     b::Fields.ColumnField,
 ) = _single_field_solve!(cache, x, A, b)
 
+# TODO: fuse
+function fused_single_field_solve!(_, x, A::UniformScaling, b)
+    foreach(matrix_row_keys(keys(A))) do name
+        single_field_solve!(cache[name], x[name], A[name, name], b[name])
+    end
+end
+fused_single_field_solve!(cache, x, A::ColumnwiseBandMatrixField, b) =
+    fused_single_field_solve!(ClimaComms.device(axes(A)), cache, x, A, b)
+
+# TODO: fuse
+function fused_single_field_solve!(
```

Comment:
Could we keep these as methods of `run_field_matrix_solver!` in `field_matrix_solver.jl`? This file should probably just have what's currently called `_single_field_solve!`; we can drop what's currently called `single_field_solve!` and move the GPU kernel logic into a fused method of `run_field_matrix_solver!`.

Reply:
Yeah, agreed, I thought about starting a new file but thought I'd see what the perf looks like first since it's a somewhat simple backend change.